web/src/main/java/org/fao/geonet/Geonetwork.java

```diff
@@ -558,6 +570,39 @@ private boolean checkDBWrite() {
         scheduledExecutorService.scheduleWithFixedDelay(DBHeartBeat, initialDelay, fixedDelay, TimeUnit.SECONDS);
     }
 
+    private void createLinkMonitor(final ResourceManager rm, final GeonetContext gc, String linkMonitorClass,
```

Comment:
I think those are fair comments. Again here, I was following the `DBHeartBeat` "example". But I might change it so that `LinkMonitorInterface` implements `Runnable`, because it must.